src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5377,9 +5377,10 @@ internal static IEnumerable<object> GetMembersByInferredType(PSTypeName typename
             List<object> results = new List<object>();
 
             Func<object, bool> filterToCall = filter;
+            var typeInferenceContext = context.TypeInferenceContext;
             if (typename.Type != null)
             {
-                if (context.CurrentTypeDefinitionAst == null || context.CurrentTypeDefinitionAst.Type != typename.Type)
+                if (typeInferenceContext.CurrentTypeDefinitionAst == null || typeInferenceContext.CurrentTypeDefinitionAst.Type != typename.Type)
```

Comment:
It looks a lot code is duplicated between here and TypeInferenceContext.cs. Duplicate code should be removed.